subprojects/core/src/main/java/org/gradle/process/internal/streams/ExecOutputHandleRunner.java

```diff
@@ -62,7 +63,7 @@ public void run() {
     }
 
     public void run(Executor executor) {
-        executor.execute(this);
+        executor.execute(new BuildOperationIdentifierPreservingRunnable(this));
```

Comment:
I think my issue with it is that not every `ExecOutputHandleRunner` needs the build operation, so we should wrap it from somewhere else.

In StreamsForwarder, we could just do:

```

    public void start() {
        executor.execute(wrapInBuildOperation(standardInputRunner));
        if (readErrorStream) {
            executor.execute(wrapInBuildOperation(errorOutputRunner));
        }
        executor.execute(wrapInBuildOperation(standardOutputRunner));
    }
    Runnable wrapInBuildOperation(Runnable runnable) {
        return new BuildOperationIdentifierPreservingRunnable(runnable);
    }
```